ui/plugin-system/src/stories/shared-utils/decorators/WithDatasourceStore.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { DatasourceSelector } from '@perses-dev/core';
+import { DatasourceStoreContext, DatasourceStore, usePluginRegistry } from '@perses-dev/plugin-system';
+import { StoryFn, StoryContext } from '@storybook/react';
+
```

Comment:
I'm not in love with this decorator, but I think it's a basic start we can incrementally improve. It's kind of a pain to work with `DatasourceStoreContext` outside of a dashboard context because we have a helpful provider for `dashboard` but not for this use case. We should probably fix that at some point, but I didn't want to deal with it right now to get this story going. I figure I hit enough levels of yak shaving already.